src/basic/efivars.c

```diff
@@ -96,6 +96,23 @@ int efi_get_variable(
                                 (void) usleep_safe(EFI_RETRY_DELAY);
                 }
 
+                /* According to comment of linux source code, efivarfs represents uncommitted
+                 * variables with zero-length files. Unfortunately kernel reports EOF if there's
+                 * an inconsistency between efivarfs var list and what's actually stored in
+                 * firmware, c.f. #34304. Hence we translate EOF back to ENOENT here, as with
+                 * kernel behavior before torvalds/linux@3fab70c.
+                 *
+                 * If the kernel changes behaviour (to flush dentries on resume), we can drop
+                 * this at some point in the future. But note that torvalds/linux@3fab70c is 11
+                 * years old at this point so we'll need to deal with the current behaviour for
+                 * a long time.
+                 *
+                 * See https://github.com/torvalds/linux/commit/3fab70c165795431f00ddf9be8b84ddd07bd1f8f
```

Comment:
Dunno. This section of information seems duplicative and can be dropped. I don't expect us to remove this in the next 5-10y if we've decided to merge this.

Reply:
It's from another reviewer

https://github.com/systemd/systemd/pull/35497#discussion_r1878443537

Maybe drop this comment is better?

@YHNdnzj @keszybz

Reply:
I'll change this comment after the decision from you two.